### PR TITLE
feat: add Whittle to Hugging Face checkpoint conversion

### PR DESCRIPTION
**Pull request by Neonkraft**
#### Reference Issues/PRs

N/A

#### What does this implement/fix? Explain your changes.

Adds `whittle.convert.save_as_hf_checkpoint`, a programmatic API for converting a Whittle super-network + sub-network config into a Hugging Face checkpoint on disk.

The function stages everything through a `TemporaryDirectory`:

1. Save the super-network as a parent LitGPT checkpoint.
2. Extract the sub-network via `save_sub_network` (format b).
3. Normalize to a plain LitGPT checkpoint via
   `whittle.convert_to_litgpt`.
4. Convert LitGPT → HF via
   `litgpt.scripts.convert_lit_checkpoint.convert_lit_checkpoint`.

The final `out_dir` ends up with just `model.pth` (HF-style state dict) and `model_config.yaml` (the sub-network's LitGPT config, kept so that callers can reconstruct the matching HF config).

Also adds `test/test_convert_to_hf.py`, parametrized across Qwen3, Llama-3, Gemma-2, and Pythia. For each model it:

- Builds a small Whittle `GPT`.
- Halves its sub-network dimensions.
- Calls `save_as_hf_checkpoint`.
- Loads the converted `model.pth` into the appropriate
  `*ForCausalLM` (`Qwen3ForCausalLM`, `LlamaForCausalLM`,
  `Gemma2ForCausalLM`, `GPTNeoXForCausalLM`).
- Asserts the HF logits match the Whittle sub-network's logits.

The second commit bumps sequence length and layer count in the test configs (`block_size=10`, `n_layer=2`) to exercise multi-token and multi-layer forward passes end-to-end.

#### Minimal Example / How should this PR be tested?

```python
from litgpt import Config
from whittle.convert import save_as_hf_checkpoint
from whittle.models.gpt import GPT

config = Config.from_name("Qwen3-0.6B", n_layer=2, n_embd=32, intermediate_size=86)
config.fix_head_size = True
model = GPT(config)

sub_network_config = {
    "embed_dim": 16,
    "mlp_ratio": 43 / 16,
    "num_heads": 8,
    "depth": 1,
    "n_query_groups": 4,
    "head_size": config.head_size,
}

save_as_hf_checkpoint(model, sub_network_config=sub_network_config, out_dir="out/")
```

Run the new test suite:

```
pytest test/test_convert_to_hf.py -v
```

#### Any other comments?

- Gemma-3 and OLMo-2 are intentionally not covered by the new test. Both hit upstream bugs in `litgpt.scripts.convert_lit_checkpoint`:
  - Gemma-3 has no dispatch branch and falls through  `copy_weights_llama`, which doesn't map `post_attention_norm` /
    `post_mlp_norm`.
  - `copy_weights_olmo2`'s weight_map references `norm_2` but the
    actual state dict key is `post_attention_norm Both can be added once LitGPT is patched.
- For Gemma-2 the test ties `lm_head` to `transformer.wte` before saving and before the comparison forward pass, matching the `tie_word_embeddings=True` convention used by the HF model.

---
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the
terms of your choice.

### CODE REVIEW

**Review by HannanJaved — APPROVED** [review]
Lgtm :)
